### PR TITLE
Added WS Connection to MessageManager

### LINKED ISSUE #113: Add web socket to availability listeners
Add web socket to message manager availability listeners when inserted into web socket manager and remove again when web socket disconnects

### PR DESCRIPTION
**Pull request by MJHC**
- Added `WebSocketConnection `as `MessageAvailabilityListener `to `MessageManager`
- Server now sends all cached and stored messages to client when they connect

closes #113

### CODE REVIEW

**Review by Mikkel-Boje — APPROVED** [review]

File: server/src/server.rs [hunk 1]
```diff
@@ -380,7 +380,19 @@ async fn create_websocket_endpoint(
...
                 state.clone(),
             );
+            let addr = ws.protocol_address();
             wmgr.insert(ws, receiver).await;
+            let ws = if let Some(x) = wmgr.get(&addr).await {
+                x
+            } else {
+                println!("ws.on_upgrade: WebSocket does not exist in WebSocketManager");
+                return;
+            };
```
Comment (t-lohse) [inline server/src/server.rs:390]:
```suggestion
            let Some(ws) = wmgr.get(&addr).await else {
                println!("ws.on_upgrade: WebSocket does not exist in WebSocketManager");
                return;
            };
```

**Review by t-lohse — CHANGES_REQUESTED** [review]
pls test this

File: server/src/server.rs (hunk 1 above)
Comment (MJHC) [reply server/src/server.rs]: ok

**Review by Mikkel-Boje — APPROVED** [review]

**Review by t-lohse — APPROVED** [review]
LGTM, thanks for the change
image

**Review by Darkros1245 — APPROVED** [review]

**Pull request merged**